Modules/Devices/src/Devices.Domain/Entities/Identities/Identity.cs

```diff
@@ -208,10 +208,11 @@ public void DeletionGracePeriodReminder3Sent()
 
     public IdentityDeletionProcess CancelDeletionProcessAsOwner(IdentityDeletionProcessId deletionProcessId, DeviceId canceledByDeviceId)
     {
+        EnsureDeletionProcessExists(deletionProcessId);
+        EnsureDeletionProcessInStatusExists(DeletionProcessStatus.Approved);
```

Comment:
I don't think that the combination of these two methods makes sense. The result could be that the deletion process with the given id exists, and there is one int status approved, but it's not inherently true that the one with the given id has the given status.

I'd suggest the following:

```c#
        var deletionProcess = GetDeletionProcessWithId(deletionProcessId); // throws if there is no deletion process with the givenId
        deletionProcess.EnsureHasStatus(DeletionProcessStatus.Approved);
```

Reply:
Changes made.